crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -198,15 +196,16 @@ impl CoinReadApiServer for CoinReadApi {
                         Some(obj) => {
                             let coin_type = obj.coin_type_maybe();
                             if coin_type.is_none() {
-                                Err(anyhow!(
-                                    "Invalid Cursor {:?}, Object is not a coin",
-                                    object_id
-                                ))
+                                Err(Error::SuiRpcInputError(SuiRpcInputError::GenericInvalid(
+                                    format!("Invalid Cursor {:?}: not a coin", object_id),
```

Comment:
nit: should we keep the object syntax here

Reply:
We shouldn't bundle the meaning of a cursor to some other entity right? From the discussion on another rpc method that returns a cursor, the fact that it's synonymous to child object id is just an artifact of the implementation. So I removed it here. But otherwise I don't feel too strongly